### DISCUSSION TITLE
[Question]: Frigate retention - cleaning up when planty of storage is left

### DISCUSSION THREAD

**Discussion by darmach** [blakeblackshear/frigate#12947] [Ask A Question] [answered]
Hi, I have found in the documentation that the recordings are moved to disk when they match the retention policy, out of the configured policies the largest is considered and recording removed if less than an hour of storage is left.

I wasn't able to find how is that determined. I would like to keep all recording, let's say, hypothetically, for 2 days, events motion for 10 days (default, not overriding).

My retention config, 4 1080p streams:
```
record:
  enabled: true
  retain:
    days: 2
    mode: all
  preview:
    # Optional: Quality of recording preview (default: medium).
    # Options are: very_low, low, medium, high, very_high
    quality: high
  events:
    pre_capture: 10
    post_capture: 10
  #   retain:
  #     default: 10
  #     mode: motion
  ```

I have noticed in the log frigate after some point running cleanup every ~1.5 hour, removing about 5G of recording - what troubles me, is that at that point I still have plenty of space left (created the VM with 64G storage from NVME pool for now).

```
2024-08-11 08:50:37.132487169  [2024-08-11 10:50:37] frigate.storage                INFO    : Less than 1 hour of recording space left, running storage maintenance...
2024-08-11 08:50:37.283494764  [2024-08-11 10:50:37] frigate.storage                INFO    : Cleaned up 4803.29999999998 MB of recordings
2024-08-11 09:45:37.299172426  [2024-08-11 11:45:37] frigate.storage                INFO    : Less than 1 hour of recording space left, running storage maintenance...
2024-08-11 09:45:37.451345051  [2024-08-11 11:45:37] frigate.storage                INFO    : Cleaned up 4800.800000000015 MB of recordings
```

![image](https://github.com/user-attachments/assets/b5a46e03-9592-451c-9010-abe60b8f2e1a)

It kind of looks like it would consider 1 hour of recording to be ~5G, multiply it by 4 (camera cound) and desire to have 20G of free space (hence cleaning after about 30 out of 50G be used. Can this be reconfigured or am I doing something wrong?

Also, a though just that just occurred to me, I should probably think about flash drive wear. Should I move the recordings pool to HDD? Is there any golden rule on how to config Frigate storage on flash devices?

**Comment by NickM-27** [comment, marked as answer]
It adds each cameras hourly storage usage to see how much storage is left. In this case it has less than an hour of storage left. That can't be disabled, and 64GB seems way too low for what you've configured the retention to be. 

I have had recordings written to a zfs pool of SSDs for multiple years now without issues.

**Reply by darmach** [reply]
Hmm, the clips are those lower res "video miniatures" introduced in 0.14 right? 
```
root@frigate:/var/lib/frigate/storage# du -sh *
9.5G    clips
45M     exports
25G     recordings
```
Judging from this, it looks like an hour of recording takes roughly 5G of space - for all four cameras together:
```
root@frigate:/var/lib/frigate/storage/recordings/2024-08-11# du -sh *
4.6G    09
4.8G    10
4.9G    11
4.8G    12
4.9G    13
926M    14
```
Indeed, you're absolutely right,  it looks like I'm running it on the edge of the freespace then!
```
root@frigate:/var/lib/frigate/storage/recordings/2024-08-11# df -h .
Filesystem                   Size  Used Avail Use% Mounted on
/dev/mapper/debian--vg-root   51G   42G  6.3G  87% /
```
I think what confused me was seemingly "a lot" of free space shown on the `system#storage` tab of the web interface.

![image](https://github.com/user-attachments/assets/162529a9-39d6-4f11-be2f-d190ff594924)

The recordings panel shows that we have about 24G free (a blatant lie! :D) what made me think that I should be able to stick 5 hours more in there. Maybe it would be good to add a kind of `Cameras - global` panel before cameras showing how much space in total for the recordings is left? Unless it's the `Recordings` panel role - then I'm afraid it's not doing a good job (showing 24G free, when we have 6.4G actually free. I wonder what is the source of this discrepancy.

So I'd have to count about 120G per day of full footage + clips additionally. I don't have enough space on nvme pool, will have to attach some old hard drive I have laying around. This raises another question - do you think I'd kill a normal consumer drive, not having some fancy NVR-ready WD Purple?

I hope I'm not an annoyance yet - this (great!) piece of software is still very new to me, and I'd like to understand inner workings well, especially when it looks I'd like to stick to it for upcoming years (coming from zoneminder).

EDIT: I noticed I nitpick a lot, and I don't want to get misunderstood - love the Frigate so far. I regret not finding it sooner! The amount of questions is result of me having a lot of fun for the past week or so! Love it!

**Reply by NickM-27** [reply]
> Hmm, the clips are those lower res "video miniatures" introduced in 0.14 right?

clips holds many different things

the reason we can't show all storage is because we don't catalog the size of the other files in /clips and checking the specific size would require spinning up the drives

I personally have not had any issues with my drives running frigate for multiple years, I have heard stories from users that said they had an SSD die but hard to say what the circumstances were exactly.

**Reply by darmach** [reply]
Hmm, I threw in a HDD into the hypervisor, added 300G to that VM, mounted under /var/lib/frigate/storage (my frigate deployment dir)

Frigate shows 300G for Recordings, but  all the cams still show 24G as max storage.

![image](https://github.com/user-attachments/assets/b6353a06-faa2-4738-a698-f2ebb2dcd70c)

**Reply by NickM-27** [reply]
it's not showing a max, it is showing out of the 24GB total, how much space is taken up by each camera

**Reply by darmach** [reply]
Ooohh, that explains a lot!
